exercise/templates/exercise/exercise.html

```diff
@@ -79,28 +79,44 @@
 <script src="{% static 'exercise/duplicate_check.js' %}"></script>
 <script src="{% static 'exercise/chapter.js' %}"></script>
 <script
-    src="{% static 'exercise/language_link.js' %}"
-    data-set-lang="{{ instance|url:'set-enrollment-language' }}"></script>
+	src="{% static 'exercise/language_link.js' %}"
+	data-set-lang="{{ instance|url:'set-enrollment-language' }}"></script>
 <script>
 // Visualize loading in case of slow POST operations.
 /*$(function() {
-    var forms = $("#exercise-page-content").find("form:not([data-aplus-exercise] form)");
-    forms.on("submit", function(event) {
-        $("#submit-progress").removeClass("hide");
-    });
+	var forms = $("#exercise-page-content").find("form:not([data-aplus-exercise] form)");
+	forms.on("submit", function(event) {
+		$("#submit-progress").removeClass("hide");
+	});
 });*/
 
 // Add an Ajax exercise event listener to refresh the info column.
 window.addEventListener("message", function (event) {
-    if (event.data.type === "a-plus-refresh-stats") {
-        $("#submit-progress").addClass("hide");
-        var $stats = $("#exercise-info"),
-            url = $stats.data("url");
-        $stats.load(url, function() {
-            $stats.find('[data-toggle="tooltip"]').tooltip();
-        });
-    }
+	if (event.data.type === "a-plus-refresh-stats") {
+		$("#submit-progress").addClass("hide");
+		var $stats = $("#exercise-info"),
+			url = $stats.data("url");
+		$stats.load(url, function() {
+			$stats.find('[data-toggle="tooltip"]').tooltip();
+		});
+	}
 });
 </script>
+
+<script>
+	//client hasn't seen the model solution tooltips
+	if (localStorage.getItem('model-solutions-tooltip-shown') !== 'true') {
+		//when ajaxStop event fires, we have received all exercises, so enable their tooltips.
+		$( document ).ajaxStop(function() {
+			const modelButtons = $('a[data-toggle="tooltip"]');
+			if (modelButtons.length) {
+				modelButtons.tooltip('show');
```

Comment:
I got these errors, which suggest that the `ajaxStop` event might not be very reliable for detecting if the document as well as all the exercises have really loaded. If Bootstrap isn't ready yet early in the page load, I guess tooltips can not work.

I was testing on the Programming 1 (O1) course.

```
Uncaught Error: cannot call methods on tooltip prior to initialization; attempted to call method 'show'
    jQuery 5
    <anonymous> http://localhost:8000/def/current/w10/ch03/#chapter-exercise-3:143
    jQuery 7
    load http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:305
    $.fn[loadName]/< http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:875
    jQuery 2
    loadName http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:872
    nextExercise http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:73
    load http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:327
    jQuery 6
    load http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:305
    $.fn[loadName]/< http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:875
    jQuery 2
    loadName http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:872
    nextExercise http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:73
    load http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:327
    jQuery 6
    load http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:305
    $.fn[loadName]/< http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:875
    jQuery 2
    loadName http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:872
    nextExercise http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:73
    load http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:327
    jQuery 6
    load http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:305
    $.fn[loadName]/< http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:875
jquery-3.6.0.min.js:2:2616
    jQuery 5
    <anonymous> http://localhost:8000/def/current/w10/ch03/#chapter-exercise-3:143
    jQuery 7
    load http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:305
    $.fn[loadName]/< http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:875
    jQuery 2
    loadName http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:872
    nextExercise http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:73
    load http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:327
    jQuery 6
    load http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:305
    $.fn[loadName]/< http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:875
    jQuery 2
    loadName http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:872
    nextExercise http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:73
    load http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:327
    jQuery 6
    load http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:305
    $.fn[loadName]/< http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:875
    jQuery 2
    loadName http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:872
    nextExercise http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:73
    load http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:327
    jQuery 6
    load http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:305
    $.fn[loadName]/< http://localhost:8000/static/exercise/chapter.js?ad188cecaa814874a5cd4bd9140ef6e195399d61:875


Uncaught Error: cannot call methods on tooltip prior to initialization; attempted to call method 'show'
    jQuery 5
    <anonymous> http://localhost:8000/def/current/w10/ch03/#chapter-exercise-3:143
    jQuery 8
    submit http://localhost:8080/static/default/_static/pheromones.js:339
    add http://localhost:8080/static/default/_static/pheromones.js:188
    addLocation http://localhost:8080/static/default/_static/pheromones.js:294
    scrollTimer http://localhost:8080/static/default/_static/pheromones.js:171
    setTimeout handler*onScroll http://localhost:8080/static/default/_static/pheromones.js:168
    bindScroll http://localhost:8080/static/default/_static/pheromones.js:180
    init http://localhost:8080/static/default/_static/pheromones.js:76
    <anonymous> http://localhost:8080/static/default/_static/pheromones.js:502
    jQuery 10
jquery-3.6.0.min.js:2:2616
```